plugins/pulp_rpm/plugins/importers/yum/modularity.py

```diff
@@ -64,7 +77,49 @@ def add_modulemd(repository, modulemd, model):
     repository_controller.associate_single_unit(repository, model)
 
 
-def add_modulemds(repository, modulemds):
+def repair_modulemd(lite_model, modulemd):
+    """
+    Repair the modulemd.
+
+    :param lite_model: A lite model fetched from inventory.
+    :type lite_model: LiteModel
+    :param modulemd: A modulemd (libmodulemd) document object.
+    :type modulemd: gi.repository.Modulemd.Module
+    :return: The repaired modulemd
+    :rtype: pulp_rpm.plugins.db.models.Modulemd
```

Comment:
It seems like this PR repairs only `modulemd` content but not `modulemd-defaults` one. Is it intentional?

Reply:
i think we do this already for the defaults every time during sync when the checksum changes no matter what's the value of the --validate flag. Because defaults are mutable, that might mean that on the remote source they got updated, meanwhile modules are immutable and are unique by NSVCA

Reply:
https://github.com/pulp/pulp_rpm/blob/2-master/plugins/pulp_rpm/plugins/importers/yum/modularity.py#L177

Reply:
@goosemania, Looking at this code: https://github.com/pulp/pulp_rpm/blob/2-master/plugins/pulp_rpm/plugins/importers/yum/modularity.py#L215 you're concern is valid.  I assumed that since the modulemd-defaults is mutable and being updated, the corruption/file-missing case was covered and it's not.  This code is comparing the checksum of the downloaded document and the checksum in the db.  I need to update the code to also compare the checksum to the stored file.